clang/lib/StaticAnalyzer/Core/CheckerManager.cpp

```diff
@@ -731,33 +731,36 @@ void CheckerManager::runCheckersForEvalCall(ExplodedNodeSet &Dst,
     ExplodedNodeSet checkDst;
     NodeBuilder B(Pred, checkDst, Eng.getBuilderContext());
 
+    ProgramStateRef State = Pred->getState();
+    CallEventRef<> UpdatedCall = Call.cloneWithState(State);
```

Comment:
Why do we need to have an updated call here?
The original Call was supposed to be wrapping the Pred state, that we wrap here again.

In other words, I think `UpdatedCall` is always the same as `Call` here. So why don't we just use `Call`?

Reply:
Yeah this may be one of _those places_. Do we really know that all the nodes in the `Src` have the same state? Because the code lowkey can't function correctly without that.